numba_cuda/numba/cuda/types.py

```diff
@@ -38,3 +39,51 @@ class CUDADispatcher(types.Dispatcher):
     # is still probably a good idea to have a separate type for CUDA
     # dispatchers, and this type might get other differentiation from the CPU
     # dispatcher type in future.
+
+
+class Bfloat16(types.Number):
+    """
+    A bfloat16 type.
+    """
+
+    def __init__(self):
+        super().__init__(name="__nv_bfloat16")
+
+        self.alignof_ = 2
+        self.bitwidth = 16
+
+    def can_convert_from(self, typingctx, other):
+        if isinstance(other, types.Float):
+            return Conversion.unsafe
+
+        elif isinstance(other, types.Integer):
+            if other.bitwidth == 8:
+                return Conversion.safe
+
+        return Conversion.unsafe
```

Comment:
If the other type is not a float or an integer, then I think it can't be converted - is that the case? If so, we should return `None` for conversions that aren't possible. Should this be:

```suggestion
            if other.bitwidth == 8:
                return Conversion.safe
            else:
                return Conversion.unsafe
```

Reply:
E.g. I think we probably don't want to be able to convert from `complex64`

Reply:
[0fa0174](https://github.com/NVIDIA/numba-cuda/pull/376/commits/0fa0174d42320b011f3fd7b3c563042d48f80c5d)